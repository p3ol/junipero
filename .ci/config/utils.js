import { mount } from 'enzyme';

export const mountToBody = component => {
  const container = document.createElement('div');
  document.body.appendChild(container);
  const wrapper = mount(component, { attachTo: container });
  const detach = wrapper.detach;
  wrapper.detach = () => {
    detach.bind(wrapper)();
    document.body.removeChild(container);
  };

  return wrapper;
};
