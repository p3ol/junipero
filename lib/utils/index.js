export const classNames = (...args) => {
  const classes = [];

  args.map((arg) => {
    if (!arg) {
      return;
    }

    if (typeof arg === 'string' || typeof arg === 'number') {
      classes.push(arg);
    } else if (Array.isArray(arg) && arg.length) {
      const inner = classNames(...arg);
      if (inner) {
        classes.push(inner);
      }
    } else if (typeof arg === 'object') {
      Object.entries(arg).map(([k, v]) => {
        if (v) {
          classes.push(k);
        }
      });
    }
  });

  return classes.join(' ');
};
