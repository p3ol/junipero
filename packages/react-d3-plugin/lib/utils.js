import { startOfDay, endOfDay } from '@junipero/react';
import * as d3 from 'd3';

export const getAxisTimeRange = (serie, opts = {}) => {
  switch (opts.granularity) {
    case 'year':
    case 'month':
      return d3.extent(serie, m => m[opts.dataIndex]);
    default:
      return d3.extent(d3.timeDay
        .range(startOfDay(opts.from), endOfDay(opts.to), 1));
  }
};

export const getAxisType = type => {
  switch (type) {
    case d3.axisLeft: return 'left';
    case d3.axisRight: return 'right';
    case d3.axisTop: return 'top';
    default: return 'bottom';
  }
};
