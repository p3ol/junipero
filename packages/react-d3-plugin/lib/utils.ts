import * as d3 from 'd3';

export const getAxisType = type => {
  switch (type) {
    case d3.axisLeft: return 'left';
    case d3.axisRight: return 'right';
    case d3.axisTop: return 'top';
    default: return 'bottom';
  }
};

export const scaleBandInvert = (scale: d3.ScaleBand<Date | number>) => {
  // Distance between the starts of adjacent bands
  const step = scale.step();
  // Space between two adjacent bands
  const paddingInner = step * scale.paddingInner();
  // Space before the first band
  const leftPad = step * scale.paddingOuter() * scale.align() * 2;
  const domain = scale.domain();

  return function (x: number): number {
    // Clamp index value between 0 and domain length
    const index = Math.min(Math.max(0, Math.floor(
      ((x - leftPad - step + paddingInner / 2) / step) + 1
    )), domain.length - 1);

    return index;
  };
};
