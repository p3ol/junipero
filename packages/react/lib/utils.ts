import type { StateContent } from './types';

export const mockState = <T extends StateContent>(
  state: T,
  action : ((prev: T) => T) | Partial<T>
) => typeof action === 'function'
    ? action(state) : ({ ...state, ...action as T });
