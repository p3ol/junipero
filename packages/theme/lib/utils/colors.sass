// Primary
$velvet: #5C56DF
$lime: #7ED496
$grapefruit: #EE7674
$sunglow: #FFB800

// Text
$onyx: #1A1A1A
$tundora: #424242

// Neutral
$seashell: #FEFEFE
$alabaster: #FCFCFC
$concrete: #F3F3F3
$mischka: #DFE1E6
$slate: #6B778C

//interface
$velvetDisabled: #D6D4F7
