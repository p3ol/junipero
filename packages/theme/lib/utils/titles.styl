@require "./vars"

h1, h2, h3, h4, h5, h6
  &.junipero
    font-weight: normal

h1.junipero
  font-size: $heading-1-font-size
  line-height: $heading-1-line-height

  &.xs
    font-size: $heading-1-font-size-xs
    line-height: $heading-1-line-height-xs

h2.junipero
  font-size: $heading-2-font-size
  line-height: $heading-2-line-height

  &.xs
    font-size: $heading-2-font-size-xs
    line-height: $heading-2-line-height-xs

h3.junipero
  font-size: $heading-3-font-size
  line-height: $heading-3-line-height

  &.xs
    font-size: $heading-3-font-size-xs
    line-height: $heading-3-line-height-xs

h4.junipero
  font-size: $heading-4-font-size
  line-height: $heading-4-line-height

  &.xs
    font-size: $heading-4-font-size-xs
    line-height: $heading-4-line-height-xs

h5.junipero
  font-size: $heading-5-font-size
  line-height: $heading-5-line-height

  &.xs
    font-size: $heading-5-font-size-xs
    line-height: $heading-5-line-height-xs

h6.junipero
  font-size: $heading-6-font-size
  line-height: $heading-6-line-height

  &.xs
    font-size: $heading-6-font-size-xs
    line-height: $heading-6-line-height-xs
