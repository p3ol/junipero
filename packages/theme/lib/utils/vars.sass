$cardShadow: 0 4px 30px -7px var(--shadow-color)
$cardShadowWithBorder: 0 0 0 1px var(--border-color), $cardShadow
