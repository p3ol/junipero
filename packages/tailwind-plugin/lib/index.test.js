import { generate } from '../tests/utils';
import plugin from './index';

describe('@junipero/tailwind-plugin', () => {
  it('should generate the correct CSS', async () => {
    const css = await generate(plugin, `
      <div class="bg-velvet"></div>
    `);

    expect(css).toMatchSnapshot();
  });
});
