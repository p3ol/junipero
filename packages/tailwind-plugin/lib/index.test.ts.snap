// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@junipero/tailwind-plugin should generate the correct CSS 1`] = `
".bg-velvet {
    --tw-bg-opacity: 1;
    background-color: rgb(92 86 223 / var(--tw-bg-opacity))
}"
`;
